sotodlib/site_pipeline/update_det_match.py

```diff
@@ -0,0 +1,374 @@
+    run_freq_offset_scan: bool
+        If true, this will scan through a range of freq-offsets (between src and
+        dst res-sets) to find an optimal offset for each match.
+    freq_offset_range_args: Optional[Tuple[float, float, float]]
+        If this is not None, for each match, we will scan over a range of
+        freq-offsets to determine the optimal offset to use. If set, must
+        contain a tuple of floats, containing ([start,] stop, [step,]) that will
+        be passed directly to ``np.arange``. If it is None, will just run with
+        the match with freq_offset_mhz=0.
+    detset_meta_name: str
+        Name of the metadata entry in the context that contains detset info.
+    detcal_meta_name: str
+        Name of the metadata entry in the context that contains det_cal info.
+    show_pb: bool
+        Will show progress bar when scanning freq-offset.
+    apply_solution_pointing: bool
+        If True, pointing information computed from design-detector positions
+        will be used in the ``merged`` detset of the match.
+    write_relpath: bool
+        If True, will use the relative path to the h5 file (relative to the db
+        path) when writing to the manifestdb
+    
+    Attriutes
+    -------------
+    freq_offsets : Optional[np.ndarray]
+        If not None, contains freq_offsets determined by
+        ``freq_offset_range_args`` which will be scanned over.
+    """
+    results_path: str
+    context_path: str
+    site_pipeline_root:str = os.environ.get('SITE_PIPELINE_CONFIG_DIR')
+    freq_offset_range_args: Optional[tuple[float, float, float]] = (-4, 4, 0.3)
+    match_pars: Optional[Dict] = None
+    detset_meta_name : str = 'smurf'
+    detcal_meta_name: str = 'det_cal'
+    show_pb: bool = False
+    apply_solution_pointing: bool = True
+    write_relpath: bool = True
+
+    def __post_init__(self):
+        if self.site_pipeline_root is None:
+            raise ValueError("Must set site_pipeline_root, or SITE_PIPELINE_CONFIG_DIR env var")
+        
+        if self.freq_offset_range_args is not None:
+            self.freq_offsets = np.arange(*self.freq_offset_range_args)
+        else:
+            self.freq_offsets = None
+
+        if self.match_pars is None:
+            self.match_pars = {}
+
+
+class Runner:
+    def __init__(self, cfg: UpdateDetMatchesConfig):
+        self.cfg = cfg
+        self.ctx = core.Context(cfg.context_path)
+        dmap_cfg_file = os.path.join(
+            cfg.site_pipeline_root, 'shared/detmapping/wafer_map.yaml'
+        )
+        with open(dmap_cfg_file, 'r') as f:
+            self.wafer_map = yaml.safe_load(f)
+        
+        self.detset_db = None
+        self.detcal_db = None
+
+        self.detmap_cfg_path = os.path.join(
+            cfg.site_pipeline_root, 'shared/detmapping/wafer_map.yaml')
+        with open(self.detmap_cfg_path, 'r') as f:
+            self.detmap_cfg = yaml.safe_load(f)
```

Comment:
This is a duplicate, somehow, of lines 99-103.

It would be nice to be able to specify this file through the cfg arg.